core/src/main/java/org/pentaho/di/connections/utils/EncryptUtils.java

```diff
@@ -39,37 +39,55 @@ public class EncryptUtils {
   private static final String SET_PREFIX = "set";
   private static final String GET_PREFIX = "get";
 
+  /**
+   * A transformer for encrypting and decrypting, which can throw a checked exception.
+   */
+  @FunctionalInterface
+  public interface Transformer<E extends Exception> {
+    /**
+     * Transforms a given string.
+     *
+     * @param value The value to transform.
+     * @return The transformed value.
+     */
+    String transform( String value ) throws E;
```

Comment:
No change needed.

I had to do some digging to understand that Encr#init is throwing the checked exception.

Reply:
Actually, there is no checked exception being thrown by either `Encr::decryptPasswordOptionallyEncrypted` or `Encr::encryptPasswordIfNotUsingVariables`. I added the checked exception to support encryption functions used by `scheduler-plugin-ee` which do throw checked exceptions.

Reply:
okay. No change needed. 

Maybe add a javadoc comment. I would hate for someone to remove the throws and then later it breaks scheduler-plugin-ee

Reply:
Addressed in https://github.com/pentaho/pentaho-kettle/pull/9171/commits/1e25408f6be64ba4e81ee697fa5127f7b64f8d89.